### ISSUE TITLE
More of a write up anywhere?

### ISSUE THREAD

**Issue by garengllc** [barbedo/vivado-git#8]
I've been looking for a good way to use git with Vivado and came across this today.  I was excited to try it today, but am having issues.  I setup the folders like you said, I create a new project via the GUI, and then when I run git init in the tcl console, it does it in the project subfolder that won't be tracked.  I am sure I am missing something stupid, but I am having issues finding a write up that walks through things in more detail than the README.  I am using Vivado 2020.1.  Are you aware of any other write ups on your tools? TIA

**Comment by garengllc** [comment]
I may have found my issue, I was creating a subfolder under vivado_project.  When I stopped doing that, the git init seems to work.  I still wouldn't mind seeing some other write ups if there are some as I am not sure I am out of the woods yet (it has been trying to do the git commit for about 20 mins so far).

**Comment by Atirat** [comment]
@garengllc @barbedo 
Hello,
did you resolve the git commit problem? I am also using Vivado 2020.1.1. It gets stuck at "common::get_param..."
I do not know what the problem is. It seems the project tcl gets generated fine and is also staged for commit but I cannot commit because Vivado gets stuck.
Any help would be appreciated.

**Comment by garengllc** [comment]
@Atirat I did get past my issues and used this setup successfully on that project.  What is the last command you are running when you are having issues?

**Comment by Atirat** [comment]
Well, as I said, I am running the "git commit" after setting up the folder structure and staging the sources. Vivado shows the commands that it is executing. The last one it shows is "common::get_param...". There it gets stuck. When I try to cancel the process it also gets stuck and I have to relaunch it. The project generation script got generated and staged for commit. I do not know, if the script got generated completely or in the right way but when I try to use it to regenerate the project it fails to do so.

**Comment by garengllc** [comment]
@Atirat When you run git commit, run it with the option to put the message with it on the command line.  The problem is that it is requesting the message, but it isn't being passed through the TCL window, so it sits waiting on the user to write one.  If you type `git commit -am "the initial commit"` is should work fine.  Let me know

**Comment by Atirat** [comment]
@garengllc Thank you very much. It worked. But now I encountered different problems related to my design. There is an AXIS Broadcaster which has an input width of 128 bit and same output widths. The generator tries to set the output width before the input width as it seems. This throws an error, because the standard input width is 8 and 128 exceeds that.

**Comment by garengllc** [comment]
Good, glad to hear it!

Unfortunately I can't help you with that new issue. I think the developer would need to chime in on that one. Good luck!

On Jul 15, 2021, 9:20 AM, at 9:20 AM, Tim ***@***.***> wrote:
***@***.*** Thank you very much. It worked. But now I encountered

**Comment by Atirat** [comment]
Well, this would be great.
One more question: Did you had to add the project summary to git as well? Because for me it would complain that it is missing if I do not add it.

**Comment by garengllc** [comment]
I have to admit that it has been a while since I've played with it.  What project summary are you referring to?  When I quickly look at my project I was using this for, I don't see one in there.

________________________________
